go-controller/pkg/util/pod_annotation.go

```diff
@@ -615,3 +625,34 @@ func AddRoutesGatewayIP(
 
 	return nil
 }
+
+// UnmarshalUDNOpenPortsAnnotation returns the OpenPorts from the pod annotation. If annotation is not present,
+// empty list with no error is returned.
+func UnmarshalUDNOpenPortsAnnotation(annotations map[string]string) ([]*OpenPort, error) {
+	result := []*OpenPort{}
+	ports, ok := annotations[UDNOpenPortsAnnotationName]
+	if !ok {
+		return result, nil
+	}
+	if err := yaml.Unmarshal([]byte(ports), &result); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal UDN open ports annotation %s: %v", ports, err)
+	}
+	allowedProtocols := sets.New("tcp", "udp", "sctp", "icmp")
```

Comment:
does this also cover ipv6?

Reply:
yes, added comment https://github.com/ovn-org/ovn-kubernetes/compare/f2fba8ee2a4fd35af8e21e6f62665cc721f1ab9d..005a822c86dcc3966c3f6396578682511b12b288